### PR TITLE
fix: Open conversation on notification click

### PR DESCRIPTION
**Pull request by borichellow**
Fix for [Task-1790](https://wearezeta.atlassian.net/browse/AR-1790)
----

# What's new in this PR?

- added Pending Intent for opening specific Conversation on notification click
- added a new `BackStackMode` for `NavigationItem` that allows us to replace an "instance of a screen" if such one is present in a backStack
- updated DeepLink handling to handle the Conversation deepLink

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-745/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-745/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-745/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3813424)(3813424)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3813424)(3813424)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 1]
```diff
@@ -103,6 +102,13 @@ class WireActivityViewModel @Inject constructor(
                         appLogger.i("IncomingCall deepLink launched from the history")
                     }
                 }
+                is DeepLinkResult.OpenConversation -> {
+                    if (!isLaunchedFromHistory(intent)) {
+                        navigationArguments.put(OPEN_CONVERSATION_ID_ARG, result.conversationsId)
+                    } else {
+                        appLogger.i("OpenConversation deepLink launched from the history")
+                    }
+                }
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:111]: to make it more readable I suggest to flip the if like:
```suggestion
                is DeepLinkResult.OpenConversation -> {
                    if (isLaunchedFromHistory(intent)) {
                    appLogger.i("OpenConversation deepLink launched from the history")
                    } else {
                        navigationArguments.put(OPEN_CONVERSATION_ID_ARG, result.conversationsId)
                    }
                }
```

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 2]
```diff
@@ -151,6 +162,10 @@ class WireActivityViewModel @Inject constructor(
         navigateTo(NavigationCommand(NavigationItem.IncomingCall.getRouteWithArgs(listOf(conversationId))))
     }
 
+    private fun openConversation(conversationId: ConversationId) {
+        navigateTo(NavigationCommand(NavigationItem.Conversation.getRouteWithArgs(listOf(conversationId)), BackStackMode.UPDATE_EXISTED))
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:166]: maybe for later: we can extract all this model creation to the navigation class or somewhere and here only pass the conversation id.

**Review by mchenani — APPROVED** [review]
couple of tiny suggestions and comments. Well done 🏋️‍♂️

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-745/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3813745)(3813745)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3813745)(3813745)-dev-release.apk)

**Review by ohassine — APPROVED** [review]
Nicely done ✅ 👌

**Pull request merged**